Sources/TSCBasic/misc.swift

```diff
@@ -224,6 +224,11 @@ public func lookupExecutablePath(
     guard let value = value, !value.isEmpty else {
         return nil
     }
+#if os(Windows)
+    let isFileName = !value.contains(":") && !value.contains("\\") && value.contains("/")
```

Comment:
This seems invalid.  `[/?<>\:*]` are not valid characters for filenames.  `[/\.?*]` at the very least is shared across all the filesystems.  `/` is a path separator, if `value.contains("/")` is true then it is most definitely not a filename.

Reply:
> `/` is a path separator, if `value.contains("/")` is true then it is most definitely not a filename.

It’s typo here. Thanks for pointing out.

Reply:
> `[/?<>\:*]` are not valid characters for path arcs. `[/\.?*]` at the very least is shared across all the filesystems.

We’re not expecting the others though. If these characters appears, I suppose any resolution could fail.
The main purpose here is to distinguish a file name from any other kinds of _valid_ path. Can you come up with some uncovered cases here?

Reply:
Technically, `\`, `/`, `:`, and `.` can appear in any _valid_ path.  A `.` can appear in a file name.  I think that this is best handled as `PathIsFileSpecW`.

Reply:
AFAIK `PathIsFileSpecW` is equivalent of `{ $0 != nil && !$0.contains(":") && !$0.contains("\\") }`. It didn’t regard `/` as path deliminator.

Reply:
Normalizing the path String can be overhead here, so I would suggest the current implementation.

Reply:
I think that we can live with the overhead for the correctness.  This is meant to be replaced with System in any case.

Reply:
The current implementation is _exactly_ correct and equal. `PathIsFileSpecW` doesn’t change its behavior ever since it was introduced.

Switching to `PathIsFileSpecW` looks a lot uglier and less clear. Also, the API is not actually following what was documented in MSDN, so we needs extra notes on it. That’s unnecessary. Why should we do that??

Reply:
I disagree on that.  I think that it is safer to work with the Win32 APIs rather than trying to re-create it.

Reply:
That's unclear (the API not works as its name suggests);
unsafe (calling C function using an unsafe buffer);
underperforming (constructing buffers and normalizing paths takes a lot more time than iterating the string itself);
unnecessary (Win32 is wrong here, while we can be correct).

Reply:
We cannot rely on Win32 specifically when it comes to paths.

I have investigated on this before I started moving to `System`. Most Win32 APIs work abnormally. They're using very easy implementation that can only fully support FAT paths. We didn't rely on it when we implement `System` either.